feat: homepage create model

av3n93rz opened the pull request:
## [ARKQB-323](https://mckinsey.atlassian.net/browse/ARKQB-323)

Adds a warning on the home page if no models are configured.
Clicking the warning the user is redirected to the Add New Model page, where they can configure their first model with an improved model configuration form, improving onboarding and user experience.
Integrates secret creation, validation, and error handling.
Refactors loading indicators and form components for consistency.

https://github.com/user-attachments/assets/4b54dc47-e0c9-4ed0-bf89-5b9fcb731d5a


#### Error states:
<img width="820" height="933" alt="Screenshot 2025-09-29 at 16 04 09" src="https://github.com/user-attachments/assets/4189832c-6724-449b-95ac-44b83ec22f9a" />
<img width="820" height="933" alt="Screenshot 2025-09-29 at 16 04 00" src="https://github.com/user-attachments/assets/55d8559f-92ad-41ae-96f9-c0ff30c1e990" />
<img width="820" height="933" alt="Screenshot 2025-09-29 at 16 03 51" src="https://github.com/user-attachments/assets/77705c61-8cc0-47d4-bf9d-64fbc09f078a" />


## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [x] Update `./docs`
- [x] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues #eg101

av3n93rz committed:
feat: Adds initial model setup form to dashboards home page

Introduces a guided setup form for configuring the first model
when none exist, improving onboarding and user experience.
Integrates secret creation, validation, and error handling.
Refactors loading indicators and form components for consistency.

av3n93rz committed:
refactor: Refactors model creation flow and improves onboarding UX

Introduces a dedicated page for adding models, removes inline model editor logic, and updates UI to guide users with no models to configure one via a visible alert. Enhances feedback, button labeling, and form clarity across model and secret creation for a smoother onboarding experience.

av3n93rz committed:
feat: Improves dashboard UI consistency and form accessibility

Refines header layout and alert styling for clearer navigation and warnings.
Updates model configuration form and secret dialog to enhance readability, accessibility, and visual consistency by removing unnecessary card wrappers and simplifying form field structures.

av3n93rz committed:
feat: Improves Select component invalid state styling

Enhances visual feedback for invalid state by updating styling classes, ensuring clearer indication for users when inputs are invalid.

av3n93rz committed:
fix: removes unnecessary button styling for spinner

Streamlines UI by eliminating extra margin and padding on the loading spinner, ensuring a cleaner appearance during pending states.

av3n93rz committed:
feat: keep showing the missing default model warning until the user creates one, prefill model creation form from query params

Enables the model configuration form to accept a default name via query parameters, streamlining the creation of a "default" model if not already present. Improves user flow by auto-populating the form when navigated from the dashboard prompt.

av3n93rz committed:
feat: improve secret creation UX and add optimistic update to useCreateSecret mutation

Enhances the model configuration form by allowing users to add new secrets directly from relevant dropdowns, with automatic form field population upon creation. Refactors secret dialog logic for better field targeting and ensures optimistic updates and rollback for secret creation.

dwmkerr commented on services/ark-dashboard/ark-dashboard/app/(dashboard)/models/page.tsx:
```diff
@@ -32,14 +32,16 @@ function ModelsContent() {
       </header>
       <div className="flex flex-1 flex-col">
-        <ModelsSection ref={modelsSectionRef} namespace={namespace} />
+        <ModelsSection namespace={namespace} />
```
I think `namespace` is no longer needed this can just be a follow up though could do it in whatever is the next PR

av3n93rz replied:
I think its fine for now, we can get rid of this when we migrate the model updates to the new form

dwmkerr commented on services/ark-dashboard/ark-dashboard/components/editors/secret-editor.tsx:
```diff
@@ -11,6 +11,7 @@ import {
 import { Input } from "@/components/ui/input"
 import { Label } from "@/components/ui/label"
 import type { Secret } from "@/lib/services/secrets"
+import { isValidK8sSecretName } from "@/lib/utils/kubernetes-validation"
```
in a future PR we can generalise this to just `isValidK8sName` cause they all have the same logic

dwmkerr commented on services/ark-dashboard/ark-dashboard/lib/utils/kubernetes-validation.ts:
```diff
@@ -45,3 +64,22 @@ export function getKubernetesNameError(name: string): string | null {
   return null;
 }
+
+export const kubernetesSecretNameSchema = z.string()
```
there's just a single validation for k8s names, not needed for this PR though

dwmkerr reviewed (changes requested):
There's a message on slack too

av3n93rz committed:
fix: corrects error toast title from 'Secret' to 'Model'

Updates the error notification to accurately reflect the entity being created, improving clarity for users encountering creation failures.

av3n93rz committed:
feat: unifies Kubernetes name validation and removes legacy logic

Replaces custom name validation functions and schemas with a single schema-based approach for Kubernetes resource names, simplifying validation logic and error messaging. Updates tests and components to use the unified schema, deprecates old helpers, and ensures consistent validation for both general and secret names.

dwmkerr committed:
Merge branch 'main' into feat/homepage_create_model
